packages/web-client/app/resources/merchant-info.ts

```diff
@@ -0,0 +1,100 @@
+import { timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { getResolver } from '@cardstack/did-resolver';
+import { Resolver } from 'did-resolver';
+import { Resource } from 'ember-resources';
+import config from '../config/environment';
+import * as Sentry from '@sentry/browser';
+
+const FIRST_RETRY_DELAY = config.environment === 'test' ? 100 : 1000;
+
+class Storage404 extends Error {}
+
+interface Args {
+  named: {
+    infoDID: string;
+    waitForInfo: boolean;
+  };
+}
+
+export class MerchantInfo extends Resource<Args> {
+  @tracked id: string | undefined;
+  @tracked name: string | undefined;
+  @tracked backgroundColor: string | undefined;
+  @tracked textColor: string | undefined;
+
+  @tracked loading = true;
+  @tracked errored: Error | undefined;
+
+  #didResolver = new Resolver(getResolver());
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.run();
+  }
+
+  private async run() {
+    try {
+      await this.fetchMerchantInfo(
+        this.args.named.infoDID,
+        this.args.named.waitForInfo
+      );
+      this.loading = false;
+    } catch (err) {
+      this.errored = err;
+      this.loading = false;
+
+      if (!(err instanceof Storage404)) {
+        console.log('Exception fetching merchant info', err);
+        Sentry.captureException(err);
+      }
+    }
+  }
+
+  private async fetchMerchantInfo(
+    infoDID: string,
+    waitForInfo = false
+  ): Promise<void> {
+    let did = await this.#didResolver.resolve(infoDID);
+    let alsoKnownAs = did?.didDocument?.alsoKnownAs;
+
+    if (alsoKnownAs) {
+      let jsonApiDocument = await this.fetchJson(alsoKnownAs[0], waitForInfo);
+      console.log('json', jsonApiDocument);
```

Comment:
I’m guessing this is from iteration in development and could be removed?

Reply:
@backspace good catch. Please push a commit to remove this and then merge this PR. :shipit: